core/src/main.cpp

```diff
@@ -267,6 +268,12 @@ constexpr auto default_jsonschema() -> std::string_view
                                         "type": "object"
                                     }},
                                     "minProperties": 1
+                                }},
+                                "mapping_lib": {{
+                                    "type": "string"
+                                }},
+                                "mapping_config": {{
```

Comment:
is this a file location?   or just settings in JSON directly?   hmmm... 

could also be ... empty.. which is a vote for JSON object (as-is)...

```suggestion
                                "user_mapping_configuration_file": {{
```

Reply:
Keeping as JSON obj unless there's a good reason otherwise.